test/TimedTestTests.h

```diff
@@ -85,6 +120,249 @@ BENCHMARK_TEST_GROUP(TimedTestTests, TimedTest)
+                           MicroBenchmarkResults::TimeType{0})
+          / DurationBench.UnsortOriginalTimings.size() };
+    TEST_WITHIN_RANGE("MicroBenchmarkDurationAverage",
+                      ExpectedAverage.count() - 1,
+                      ExpectedAverage.count() + 1,
+                      DurationBench.Average.count());
+
+    TEST_WITHIN_RANGE("MicroBenchmarkDurationFastest",
+                      Pentile1TimeLower.count(),
+                      Pentile1TimeUpper.count(),
+                      DurationBench.Fastest.count());
+
+    TEST_WITHIN_RANGE("MicroBenchmarkDurationPercentile1st",
+                      Pentile1TimeLower.count(),
+                      Pentile1TimeUpper.count(),
+                      DurationBench.FasterThan99Percent.count());
+
+    TEST_WITHIN_RANGE("MicroBenchmarkDurationPercentile10th",
+                      Pentile1TimeLower.count(),
+                      DurationBench.GetIndexValueFromPercent(0.2).count(),
+                      DurationBench.FasterThan90Percent.count());
+
+    TEST_WITHIN_RANGE("MicroBenchmarkDurationMedian",
+                      DurationBench.GetIndexValueFromPercent(0.4).count(),
+                      DurationBench.GetIndexValueFromPercent(0.6).count(),
+                      DurationBench.Median.count());
+
+    TEST_WITHIN_RANGE("MicroBenchmarkDurationPercentile90th",
+                      Pentile5TimeLower.count(),
+                      DurationBench.FasterThan1Percent.count(),
+                      DurationBench.FasterThan10Percent.count());
+
+    TEST("MicroBenchmarkDurationPercentile99th", Pentile5TimeLower.count()< DurationBench.FasterThan1Percent.count());
+    TEST("MicroBenchmarkDurationSlowest", Pentile5TimeLower.count() < DurationBench.Slowest.count());
+
+    // This is a halfway reasonable example to compare two potentially close performing tasks. In your tests you should
+    // never use a hardcoded number because any number of factors could change it. Rather generate two measurements and
+    // and somehow compare those.
+
+    // These represent two different algorithms to check. Actually having both in tests lets us check them as platform,
+    // cache situations, type implementations, compilers and any other conditions.
+    auto FastThingToCheck = []{ std::this_thread::sleep_for(std::chrono::milliseconds{10}); };
+    auto SlowThingToCheck = []{ std::this_thread::sleep_for(std::chrono::milliseconds{20}); };
+
+    // Do 1000 iterations of each so the values are statistically significant
+    const MicroBenchmarkResults FastMeasurements = MicroBenchmark(5000, std::move(FastThingToCheck));
+    const MicroBenchmarkResults SlowMeasurements = MicroBenchmark(5000, std::move(SlowThingToCheck));
+
+    // Are all but the worst of the faster one faster than all but the best of best of the slower algorithm
+    TEST("ExampleAlgorithmComparis", FastMeasurements.FasterThan10Percent < SlowMeasurements.FasterThan90Percent);
```

Comment:
Comparis?

Reply:
Fixed, and I revised all the comments in this area.